nb/projects/belt_sprockets/belt_sprockets.nb

```diff
@@ -0,0 +1,741 @@
+//Ported to JSxCAD from https://www.thingiverse.com/thing:16627/files by @BarbourSmith
+
+// Parametric Pulley with multiple belt profiles
+// by droftarts January 2012
+
+// Based on pulleys by:
+// http://www.thingiverse.com/thing:11256 by me!
+// https://github.com/prusajr/PrusaMendel by Josef Prusa
+// http://www.thingiverse.com/thing:3104 by GilesBathgate
+// http://www.thingiverse.com/thing:2079 by nophead
+
+// dxf tooth data from http://oem.cadregister.com/asp/PPOW_Entry.asp?company=915217&elementID=07807803/METRIC/URETH/WV0025/F
+// pulley diameter checked and modeled from data at http://www.sdp-si.com/D265/HTML/D265T016.html
+
+// tune-able constants
+
+//  ********************************
+//  ** Scaling tooth for good fit **
+//  ********************************
+/*  To improve fit of belt to pulley, set the following constant. Decrease or increase by 0.1mm at a time. We are modelling the *BELT* tooth here, not the tooth on the pulley. Increasing the number will *decrease* the pulley tooth size. Increasing the tooth width will also scale proportionately the tooth depth, to maintain the shape of the tooth, and increase how far into the pulley the tooth is indented. Can be negative */
+
+const additionalToothWidth = 0.2; //mm
+
+//  If you need more tooth depth than this provides, adjust the following constant. However, this will cause the shape of the tooth to change.
+
+const additionalToothDepth = 0; //mm
+
+// Functions
+
+const ToothSpaceingCurvefit = (teeth, b, c, d) => {
+  return ((c * Math.pow(teeth, d)) / (b + Math.pow(teeth, d))) * teeth;
+};
+
+const ToothSpacing = (teeth, tooth_pitch, pitch_line_offset) =>
+  2 * ((teeth * tooth_pitch) / (3.14159265 * 2) - pitch_line_offset);
+
+const ToothProfileToFace = (...points) =>
+  Shape.fromOpenPath(points).close().fill();
+
+const MXL = {
+  profile: ToothProfileToFace(
```

Comment:
The profile shape won't get cached while it's in here, which I think isn't a problem.

But if you wanted it to be cached you could restructure it as

const xmlProfile = ToothProfileToFace(...);

const xml = { profile: xmlProfile, ... };

Reply:
I feel like its a small enough bit of memory that it's not a big concern, but its good to keep in mind

Reply:
sg